code/utils/Random.cpp

```diff
@@ -0,0 +1,76 @@
+#include "Random.h"
+
+#include <limits>
+#include <random>
+
+#include <cstdint>
+
+#include "globalincs/pstypes.h"
+
+namespace util {
+namespace {
+template <typename RngType>
+class RandomImpl {
+public:
+void seed(const unsigned int val) {
+	static_assert(RngType::min() == 0, "RNG min must be 0");
+	static_assert(RngType::max() == std::numeric_limits<uint32_t>::max(), "RNG max must be INT32_MAX");
+
+	// follow srand semantics
+	if (val == 1) {
+		m_rng.seed();
+	} else {
+		m_rng.seed(val);
+	}
+}
+
+int next()
+{
+	return m_rng() & Random::MAX_VALUE;
+}
+
+void advance(unsigned long long distance)
+{
+	m_rng.discard(distance);
+}
+
+private:
+	RngType m_rng;
+};
+
+RandomImpl<std::mt19937> SCP_rng;
+} // namespace
+
+Random::Random() = default;
+
+void Random::seed(unsigned int val)
+{
+	Assert(val > 0);
+	SCP_rng.seed(val);
+}
+
+int Random::next()
+{
+	return SCP_rng.next();
+}
+
+int Random::next(int modulus)
+{
+	Assert(modulus > 0);
+
+	return next() % modulus;
+}
+
+int Random::next(int low, int high)
+{
+	const int range = high - low + 1;
```

Comment:
This `const` is pointless since the variable is created and destroyed across function calls.

Reply:
No it's not. Marking variables as `const`is very useful for ensuring that later changes don't accidentally change previously constant values.

Reply:
Agreed with @asarium about the general principle, but I'm not gonna make a fuss over this one. If the team (or a majority) wants the `const` removed, I'll drop it.